diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +119,200 @@ class LineLength : Rule("line-length") {
+    }
+
+    /**
+     * This method stored all the nodes that have BINARY_EXPRESSION element type.
+     * This method uses recursion to store binary node in the order in which they are located
+     * Also binList contains nodes with PREFIX_EXPRESSION element type ( !isFoo(), !isValid)
+     *@param node node in which to search
+     *@param binList mutable list of ASTNode to store nodes
+     */
+    private fun searchBinaryExpression(node: ASTNode, binList: MutableList<ASTNode>) {
+        when {
+            node.hasChildOfType(BINARY_EXPRESSION) -> {
+                node.getAllChildrenWithType(BINARY_EXPRESSION).forEach {
+                    searchBinaryExpression(it, binList)
+                }
+            }
+            node.hasChildOfType(PARENTHESIZED) -> {
+                node.getAllChildrenWithType(PARENTHESIZED).forEach {
+                    searchBinaryExpression(it, binList)
+                }
+            }
+            else -> {
+                binList.add(node)
+                binList.add(node.treeParent.findChildByType(PREFIX_EXPRESSION) ?: return)
+            }
+        }
+    }
+
+    private fun dfsForProperty(node: ASTNode, binList: MutableList<ASTNode>) {
+        node.getChildren(null).forEach {
+            if (it.elementType in PROPERTY_LIST)
+                binList.add(it)
+            dfsForProperty(it, binList)
+        }
+    }
+
+    private fun fixProperty(parent: ASTNode, configuration: LineLengthConfiguration){
+        if (!parent.hasChildOfType(STRING_TEMPLATE)) {
+            if (parent.hasChildOfType(BINARY_EXPRESSION)) {
+                val leftOffset = positionByOffset(parent.findChildByType(BINARY_EXPRESSION)!!.psi.textOffset).second
+                fixCondition(parent, configuration, leftOffset, true)
+            }
+            if (parent.hasChildOfType(PARENTHESIZED)){
+                val newParent = parent.findChildByType(PARENTHESIZED)!!
+                val leftOffset = positionByOffset(newParent.findChildByType(BINARY_EXPRESSION)!!.psi.textOffset).second
+                fixCondition(newParent, configuration, leftOffset)
+            }
+        }
+        else
+            createNodes(parent, configuration)
+    }
+
+    private fun createNodes(parent: ASTNode, configuration: LineLengthConfiguration) {
+        var text = parent.findChildByType(STRING_TEMPLATE)!!.text
+        text = text.trimStart(text.first()).trimEnd(text.last())
+        val col = positionByOffset(parent.findChildByType(STRING_TEMPLATE)!!.psi.textOffset).second
+        parent.removeChild(parent.findChildByType(STRING_TEMPLATE)!!)
+        val correctTextList = mutableListOf<String>()
+        correctTextList.add(text.substring(0, configuration.lineLength.toInt() - col - FIRST_STRING_PART_OFFSEST))
+        text = text.substring(configuration.lineLength.toInt() - col - FIRST_STRING_PART_OFFSEST)
+        correctTextList.addAll(text.chunked(configuration.lineLength.toInt() - REST_STRING_PART_OFFEST))
```

Comment:
```suggestion
        val correctTextList = mutableListOf<String>().apply {
            add(text.substring(0, configuration.lineLength.toInt() - col - FIRST_STRING_PART_OFFSEST))
            text = text.substring(configuration.lineLength.toInt() - col - FIRST_STRING_PART_OFFSEST)
            addAll(text.chunked(configuration.lineLength.toInt() - REST_STRING_PART_OFFEST))
        }
        .toList()
```
If you need mutability only to fill collection with values, it's good practice to wrap it inside `apply` or similar function and then convert to immutable collection.